src/streamer.cpp

```diff
@@ -1063,17 +1063,23 @@ void Streamer::discoverPickups(Player &player, const std::vector<SharedCell> &ce
 	{
 		for (boost::unordered_map<int, Item::SharedPickup>::const_iterator p = (*c)->pickups.begin(); p != (*c)->pickups.end(); ++p)
 		{
+			if (p->second->worlds.empty())
+			{
+				p->second->worlds.insert(-1);
```

Comment:
I've not tested this, but I'm pretty sure that explicitly adding `-1` will interfere with the array data manipulation functions. An empty container is supposed signify that the pickup exists in all worlds.

Reply:
My bad, I shouldn't have changed it. Should be better now.